cvprac/cvp_api.py

```diff
@@ -3204,6 +3210,449 @@ def workspace_build_status(self, workspace_id, build_id):
+        else:
+            params = 'key.id={}&time={}'.format(cc_id, cc_time)
+        url = '/api/resources/changecontrol/v1/ApproveConfig?' + params
+        # For on-prem check the version as it is only supported from 2021.2.0+
+        if not self.clnt.is_cvaas:
+            if self.clnt.apiversion is None:
+                self.get_cvp_info()
+            if self.clnt.apiversion < 6.0:
+                self.log.warning('change_control_approval_get_one: Change Control Resource APIs are supported'
+                                    ' from 2021.2.0 or newer.')
+                return None
+        try:
+            cc_status = self.change_control_get_one(cc_id)
+        except Exception as e:
+            self.log.warning('change_control_get_one: The change control ID does not exist.')
+            return None
+        if 'approve' not in cc_status['value']:
+            self.log.warning("change_control_approval_get_one: The change has not been approved yet."
+                                " A change has to be approved at least once for the 'approve' state to be populated.")
+            return None
+        return self.clnt.get(url, timeout=self.request_timeout)
+
+
+
+    def change_control_approval_get_all(self):
+        ''' Get state information for all Change Control Approvals using Resource APIs.
+            Supported versions: CVP 2021.2.0 or newer and CVaaS.
+
+            Returns:
+               response (dict): A dict that contains a list of all Change Control Approval Configs.
+        '''
+        url = '/api/resources/changecontrol/v1/ApproveConfig/all'
+        # For on-prem check the version as it is only supported from 2021.2.0+
+
+        if not self.clnt.is_cvaas:
+            if self.clnt.apiversion is None:
+                self.get_cvp_info()
+            if self.clnt.apiversion < 6.0:
+                self.log.warning('change_control_approval_get_all: Change Control Resource APIs are supported'
+                                     ' from 2021.2.0 or newer.')
+                return None
+        self.log.debug('v6 {}'.format(url))
+        return self.clnt.get(url, timeout=self.request_timeout)
+
+    def change_control_approve(self, cc_id, notes="", approve=True):
+        ''' Approve/Unapprove a change control using Resource APIs.
+            Supported versions: CVP 2021.2.0 or newer and CVaaS.
+
+            Args:
+              cc_id (str): The ID of the change control.
+              notes (str): An optional approval note.
+              approve (bool): Set to True to approve a change and to False to unapprove a change. The default is True.
+        '''
+        url = '/api/resources/changecontrol/v1/ApproveConfig'
+        # For on-prem check the version as it is only supported from 2021.2.0+
+        # Since the get_change_control already checks this, no need to check it again
+        try:
+            version = self.change_control_get_one(cc_id)['value']['change']['time']
```

Comment:
Check response from change_control_get_one to assign version instead of assuming the keys 'value', 'change', and 'time' will be in the response data.